src/main/kotlin/at/bitfire/dav4jvm/DavResource.kt

```diff
@@ -151,22 +151,30 @@ open class DavResource @JvmOverloads constructor(
 
     /**
      * Sends an OPTIONS request to this resource without HTTP compression (because some servers have
-     * broken compression for OPTIONS). Doesn't follow redirects.
+     * broken compression for OPTIONS). Follows up to [MAX_REDIRECTS] redirects when set.
      *
+     * @param followRedirects whether redirects should be followed (default: *false*)
      * @param callback called with server response unless an exception is thrown
      *
      * @throws IOException on I/O error
      * @throws HttpException on HTTP error
      * @throws DavException on HTTPS -> HTTP redirect
      */
     @Throws(IOException::class, HttpException::class)
-    fun options(callback: CapabilitiesCallback) {
-        httpClient.newCall(Request.Builder()
+    fun options(followRedirects: Boolean = false, callback: CapabilitiesCallback) {
+        val callBlock = {
```

Comment:
I'd call it "doOptions" or "sendRequest" or something like that (to express what it actually does).